### PR TITLE
feat: install local engine

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes

Syntax:
`-s, --source`: local folder source to install engine
```
cortex engines install <engine_name> -s <source_path>
```

- If source_path contains engine → unzip engine to installation directory
- If source_path does not contain engine → fallback to download engine path

The same logic for CUDA dependencies
## Fixes Issues

- https://github.com/janhq/cortex.cpp/issues/1217

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/services/engine_service.h [hunk 1]
```diff
@@ -1,9 +1,12 @@
 #pragma once
 
+#include <memory>
 #include <optional>
 #include <string>
 #include <string_view>
 #include <vector>
+#include "utils/cpuid/cpu_info.h"
+// #include "utils/system_info_utils.h"
```
Comment (namchuai) [inline engine/services/engine_service.h]: nit: remove this comment include

File: engine/services/engine_service.cc [hunk 2]
```diff
@@ -282,23 +285,100 @@ void EngineService::InstallEngine(const std::string& engine,
...
+  if (semantic_version_utils::CompareSemanticVersion(
+          hw_inf_.cuda_driver_version, suitable_toolkit_version) < 0) {
+    CTL_ERR("Your Cuda driver version "
+            << hw_inf_.cuda_driver_version
+            << " is not compatible with cuda toolkit version "
+            << suitable_toolkit_version);
+    throw std::runtime_error("Cuda driver is not compatible with cuda toolkit");
   }
+
+  std::ostringstream cuda_toolkit_url;
```
Comment (namchuai) [inline engine/services/engine_service.cc]: nit: can use our `url_parser`

**Review by namchuai — APPROVED** [review]
✅ some small nits. Looks good to me overally!

**Review by nguyenhoangthuan99 — APPROVED** [review]

**Review by dan-menlo — APPROVED** [review]
Lgtm, thanks @vansangpfiev. A few follow-up questions: 

- For `--source`, will it take a folder or a file?
- Is it a folder since CUDA dependencies and llamacpp tar.gz are different files?

I will let the engineers scrutinize for code convention.

**Comment by vansangpfiev** [comment]
> Lgtm, thanks @vansangpfiev. A few follow-up questions:
> 
> * For `--source`, will it take a folder or a file?
> * Is it a folder since CUDA dependencies and llamacpp tar.gz are different files?
> 
> I will let the engineers scrutinize for code convention.

It is a folder

**Pull request merged**